modules/storage-api/src/main/java/org/apache/ignite/internal/storage/InvokeClosure.java

```diff
@@ -17,14 +17,27 @@
 
 package org.apache.ignite.internal.storage;
 
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-/** */
-public interface InvokeClosure {
+/**
+ * Closure that performs an operation on the storage.
+ *
+ * @param <T> Type of the invocation's result.
+ */
+public interface InvokeClosure<T> {
+    /**
+     * @param row Old row.
+     */
+    void call(@NotNull DataRow row);
+
     /**
-     * @param row Old row or {@code null} if the old row has not been found.
+     * @return Result of the invocation. Can be {@code null}.
      */
-    void call(@Nullable DataRow row);
+    @Nullable
+    default T result() {
```

Comment:
why do you need this method, why can't `call` return `T`?

Reply:
The invocation of `call` doesn't really do anything, so I decided that it's better to use separate method for the result